manifests/init.pp

```diff
@@ -107,6 +108,12 @@
     validate_string($npm_package_name)
   }
 
+  if $npmrc_auth {
+    if is_string($npmrc_auth) == false {
```

Comment:
Did this instead of validate_string because I wanted to check for the Puppet::Error and a specific error string which can change quite a bit using `validate_string()` on different platforms.